python/unblob/handlers/compression/geom_uzip.py

```diff
@@ -0,0 +1,79 @@
+from io import SEEK_SET
+from pathlib import Path
+from typing import Optional
+from zlib import decompress
+
+from structlog import get_logger
+
+from unblob.file_utils import Endian, FileSystem, StructParser, read_until_past
+from unblob.models import (
+    Extractor,
+    ExtractResult,
+    File,
+    HexString,
+    StructHandler,
+    ValidChunk,
+)
+
+""",
+The geom_uzip header follows the following structure:
+10 bytes shebang, with newline suffix: #!/bin/sh\n
+13 bytes version, with newline suffix: #V2.0 Format\n or #L3.0 Format\n
+105 bytes command, with null bytes suffix: (kldstat -qm g_uzip||kldload geom_uzip)>&-&&mount_cd9660 /dev/`mdconfig -af $0`.uzip $1\nexit $?\n\x00\x00\x00\x00\x00\x00\x00\x00\x00
+"""
+
+C_DEFINITIONS = r"""
+    typedef struct uzip_header{
+        char magic[10];             /* '10 bytes '*/
+        char version[13];           /* 13 bytes */
+        char format[105];           /* 105 bytes */
+        uint32_t block_size;        /* 4 bytes */
+        uint32_t block_count;       /* 4 bytes - Number of blocks */
+        uint64_t toc[block_count];  /* table of content */
+    } uzip_header_t;
+"""
+HEADER_STRUCT = "uzip_header_t"
+
+logger = get_logger()
+
+
+class UZIPExtractor(Extractor):
+    def extract(self, inpath: Path, outdir: Path):
+        infile = File.from_path(inpath)
+        parser = StructParser(C_DEFINITIONS)
+        header = parser.parse(HEADER_STRUCT, infile, Endian.BIG)
+        fs = FileSystem(outdir)
+        outpath = Path(inpath.stem)
+        with fs.open(outpath, "wb+") as outfile:
+            for current_offset, next_offset in zip(header.toc[:-1], header.toc[1:]):
+                compressed_len = next_offset - current_offset
+                if compressed_len == 0:
+                    continue
+                infile.seek(current_offset, SEEK_SET)
+                outfile.write(decompress(infile.read(compressed_len)))
```

Comment:
This makes it more obvious that we're decompressing zlib streams here:

```suggestion
                outfile.write(zlib.decompress(infile.read(compressed_len)))
```

Reply:
ok, I just didn't wanted to import the entire lib